### PR TITLE
Only dedupe photos by dateTaken, not datePublished

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

**Comment by aboodman** [comment]
Fixes #2723

File: samples/js/photos/src/photo-grid.js [hunk 1]
```diff
@@ -155,15 +155,18 @@ export default class PhotoGrid extends React.Component<void, Props, State> {
...
     const moreP = [];
-    let lastNegdate = -Infinity;
+    let lastTaken = Infinity;
     let next;
     while (!(next = await photosIter.next()).done && moreP.length < photosPerPage) {
-      const [negdate, nomsPhoto] = notNull(next.value);
-      if ((negdate - lastNegdate) > timeGroupThresholdMs) {
+      const [negDate, nomsPhoto] = notNull(next.value);
+      const taken = nomsPhoto.dateTaken;
+      if (!taken || ((lastTaken - taken.nsSinceEpoch) > timeGroupThresholdMs)) {
```
Comment (kalman) [inline samples/js/photos/src/photo-grid.js]: comment? this stuff is a bit subtle.

**Comment by kalman** [comment]
Though flow errors. You'll need to update types.js.

**Comment by aboodman** [comment]
done and done

**Pull request merged**